File: Tests/SwiftKotlinFrameworkTests/Tests/statics.kt

class A {
    companion object {
        public var myBool = true
    }
}
class A {
    companion object {
        private var myBool = true
        var myNum = 3
        var myString = "string"
    }
}
class A {
    companion object {
        fun method() {}
    }
}
class A {
    companion object {
        fun method() {}
        fun create() : A? {
            return null
        }
        fun withParams(param: Int) : A? {
            return null
        }
    }
}
class A {
    companion object {
        var myBool = true
        fun method() {}
    }

    var name = "string"
    fun test() {}
}
data class A {
    companion object {
        var myBool = true
        fun method() {}
    }

    var name = "string"
    fun test() {}
}

File: Tests/SwiftKotlinFrameworkTests/Tests/statics.swift

class A {
    public static var myBool = true
}

class A {
    static private var myBool = true
    static var myNum = 3
    static var myString = "string"
}

class A {
    static func method() {}
}

class A {
    static func method() {}
    static func create() -> A? { return nil }
    static func withParams(param: Int) -> A? { return nil }
}

class A {
    var name = "string"
    static var myBool = true
    static func method() {}
    func test() {}
}

struct A {
    var name = "string"
    static var myBool = true
    static func method() {}
    func test() {}
}
